fuel/transformers/__init__.py

```diff
@@ -344,12 +431,20 @@ class Batch(Transformer):
         raised if a batch of the requested size cannot be provided.
 
     """
-    def __init__(self, data_stream, iteration_scheme, strictness=0):
+    def __init__(self, data_stream, iteration_scheme, strictness=0, **kwargs):
+        if not data_stream.produces_examples:
+            raise ValueError('the wrapped data stream must produce examples, '
+                             'not batches of examples.')
+        if data_stream.axis_labels:
+            kwargs.setdefault(
+                'axis_labels',
+                dict([(source, ('batch',) + labels if labels else None) for
+                      source, labels in iteritems(data_stream.axis_labels)]))
```

Comment:
Not sure I agree: the `AbstractDataStream` setter for `produces_examples` will complain because there's an iteration scheme for this stream.

I think we should check that the iteration scheme requests batches, though.